src/main/java/com/project/videotecha/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package com.project.videotecha.service.impl;
+
+import com.project.videotecha.model.User;
+import com.project.videotecha.repository.UserRepository;
+import com.project.videotecha.service.AuthenticationService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private final UserRepository userRepository;
+
+    public AuthenticationServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public User registerUser(User newUser) {
+        userRepository.save(newUser);
+        return userRepository.findByEmail(newUser.getEmail());
```

Comment:
`save` method returns the created user, there is no need to make another call to the db by calling `findByEmail`.